python/speech-to-text/main.py

```diff
@@ -0,0 +1,83 @@
+import os
+import json
+import wave
+import base64
+from googleMain import google_speech_recognition
+from amazonMain import amazon_recognize
+from azureSpeech import azure_recognize
+
+def write_wav(audioBase64, fileName):
+    base_path = os.path.dirname(os.path.abspath(__file__))
+    filePath = os.path.join(base_path, "tempAudio",fileName)
+
+    try :
+        wav_file = open(filePath, "wb")
+        decode_string = base64.b64decode(audioBase64)
+        wav_file.write(decode_string)
+
+        obj = wave.open(filePath, 'r')
+        channels = obj.getnchannels()
+        rate = obj.getframerate()
+    except Exception as e:
+        raise Exception(f'Error in processing wav file from base64: {str(e)}')
```

Comment:
best practice when raising an exception in an except block is to add `from <original exception`:

```suggestion
        raise Exception(f'Error in processing wav file from base64: {str(e)}') from e
```